go-controller/pkg/ovn/controller/apbroute/external_controller_namespace_test.go

```diff
@@ -178,254 +221,299 @@ var _ = Describe("OVN External Gateway namespace", func() {
+					nil,
+					&v1.LabelSelector{MatchLabels: gatewayNamespaceMatch},
+					&v1.LabelSelector{},
+					false,
+				)
+				initController([]runtime.Object{namespaceTarget2, targetPod2, namespaceGW, podGW, annotatedPodGW}, []runtime.Object{dynamicPolicy})
+				namespaceGWWith2Pods := newNamespaceWithPods(namespaceGW.Name, podGW, annotatedPodGW)
+				policyName := dynamicPolicy.Name
+
+				expectedPolicy, expectedRefs := expectedPolicyStateAndRefs(
+					[]*namespaceWithPods{namespaceTarget2WithPod},
+					nil,
+					[]*namespaceWithPods{namespaceGWWith2Pods}, false)
 
 				By("validating that the namespace cache contains the test namespace and that it reflect the applicable policy")
-				Eventually(func() []string { return listNamespaceInfo() }, 5).Should(HaveLen(1))
-				Eventually(func() *namespaceInfo { return getNamespaceInfo(namespaceTest.Name) }, 5).Should(
-					BeComparableTo(
-						&namespaceInfo{
-							Policies:       sets.New(dynamicPolicy.Name),
-							StaticGateways: gatewayInfoList{},
-							DynamicGateways: map[ktypes.NamespacedName]*gatewayInfo{
-								{Namespace: podGW.Namespace, Name: podGW.Name}: newGatewayInfo(sets.New(dynamicHopHostNetPodIP), false)}},
-						cmpOpts...))
+				eventuallyExpectNumberOfPolicies(1)
+				eventuallyExpectConfig(policyName, expectedPolicy, expectedRefs)
 			})
 
-			It("deletes an existing namespace with one policy and no pods hosted in and then creates it again and validates the policy has been applied to the new one with equal values", func() {
-				expected := &namespaceInfo{
-					Policies:       sets.New(dynamicPolicy.Name),
-					StaticGateways: gatewayInfoList{},
-					DynamicGateways: map[ktypes.NamespacedName]*gatewayInfo{
-						{Namespace: podGW.Namespace, Name: podGW.Name}: newGatewayInfo(sets.New(dynamicHopHostNetPodIP), false)}}
-
-				initController([]runtime.Object{namespaceTest, namespaceDefault, podGW, annotatedPodGW}, []runtime.Object{dynamicPolicy})
-				Eventually(func() []string { return listNamespaceInfo() }, 5).Should(HaveLen(1))
-				Eventually(func() *namespaceInfo { return getNamespaceInfo(namespaceTest.Name) }, 5).Should(BeComparableTo(expected, cmpOpts...))
-				deleteNamespace(namespaceTest.Name, fakeClient)
-				By("validating that the namespace cache no longer contains the test namespace")
-				Eventually(func() []string { return listNamespaceInfo() }, 5).Should(HaveLen(0))
-
-				createNamespace(namespaceTest)
-				By("validating that the namespace cache is contained in the namespace info cache and it reflects the correct policy")
-
-				Eventually(func() []string { return listNamespaceInfo() }, 5).Should(HaveLen(1))
-				Eventually(func() *namespaceInfo { return getNamespaceInfo(namespaceTest.Name) }, 5).Should(BeComparableTo(expected, cmpOpts...))
+			It("deletes an existing namespace with one policy and then creates it again and validates the policy has been applied to the new one with equal values", func() {
```

Comment:
maybe the original one should have been defined in the `delete` context section instead, what do you think about moving it there?